modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheDefaultAffinityKeyMapper.java

```diff
@@ -77,6 +78,9 @@
     @Override public Object affinityKey(Object key) {
         GridArgumentCheck.notNull(key, "key");
 
+        if(key.getClass().isArray())
+            return IgniteUtils.hashCode(key);
```

Comment:
@ingvard It is puzzling for me why an _affinity key_ for an array is calculated as a _hash_ for an array content? Could you please elaborate?